docs/rules/jsx-tag-spacing.md

```diff
@@ -0,0 +1,179 @@
+# Validate whitespace in and around the JSX opening and closing brackets (jsx-tag-spacing)
+
+Enforce or forbid spaces after the opening bracket, before the closing bracket of self-closing elements, and between the angle bracket and slash of JSX closing or self-closing elements.
+
+## Rule Details
+
+This rule checks the whitespace inside and surrounding the JSX syntactic elements.
+
+This rule takes one argument, an object with 3 possible keys: `closingSlash`, `beforeSelfClosing` and `afterOpening`. Each key can receive the value `"allow"` to disable that specific check.
+
+The default values are:
+
+```json
+{
+  "closingSlash": "never",
+  "beforeSelfClosing": "always",
+  "afterOpening": "never"
+}
+```
+
+The options for each sub-option are documented in the following subsections.
+
+### `closingSlash`
+
+This check can be set to `"always"`, `"never"` or `"allow"` (to disable it).
+
+If it is `"never"`, the check warns whenever a space is separating the two characters in the JSX tokens `</` and `/>`. If it is `"always"` then it warns whenever a space is missing separating the mentioned two characters. The default value of this check is `"never"`.
+
+The following patterns are considered warnings when configured `"never"`:
+
+```js
```

Comment:
It would be great if these used `jsx` instead of `js` for better syntax highlighting.